atlasaction/action.go

```diff
@@ -24,15 +25,51 @@ import (
 	"ariga.io/atlas/sql/sqlcheck"
 	"github.com/mitchellh/mapstructure"
 	"github.com/sethvargo/go-envconfig"
-	"github.com/sethvargo/go-githubactions"
 )
 
 // Version holds atlas-action version. When built with cloud packages should be set by build flag, e.g.
 // "-X 'ariga.io/atlas-action/atlasaction.Version=v0.1.2'"
 var Version string
 
+// Atlas action interface.
+type Action interface {
+	Logger
+	// GetInput returns the value of the input with the given name.
+	GetInput(string) string
+	// SetOutput sets the value of the output with the given name.
+	SetOutput(string, string)
+	// TriggerContext returns the context of the trigger event.
+	GetTriggerContext() (*TriggerContext, error)
+	// AddStepSummary adds a summary to the action step.
+	AddStepSummary(string)
+}
+
+type Logger interface {
+	// Infof logs an info message.
+	Infof(string, ...interface{})
+	// Warningf logs a warning message.
+	Warningf(string, ...interface{})
+	// Errorf logs an error message.
+	Errorf(string, ...interface{})
+	// Fatalf logs a fatal error message and exits the action.
+	Fatalf(string, ...interface{})
+	// WithFieldsMap returns a new Logger with the given fields.
+	WithFieldsMap(map[string]string) Logger
+}
+
+// Context holds the context of the environment the action is running in.
+type TriggerContext struct {
+	Repository string
+	Event      map[string]interface{}
+	EventName  string
+	HeadRef    string
+	RefName    string
+	APIURL     string
+	SHA        string
+}
```

Comment:
I think this structure should include all information for `githubTriggerEvent`. And remove `githubTriggerEvent` structure.

Reply:
`EventName string` can replace with `HasPullRequest() bool`

Reply:
`APIURL     string` can be replace with `os.Getenv("GITHUB_SERVER_URL")` so it will work with other environment

Reply:
I think I will move these changes in a another PR with better context, right now the PR focus on reducing the dependent on github action